src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1271,16 +1281,23 @@ private void Abort(Exception abortException)
 
         private void AbortStreams(int lastValidStream, Exception abortException)
         {
-            bool shouldInvalidate = false;
-            lock (SyncObject)
+            Debug.Assert(lastValidStream >= 0);
+            bool hasAnyActiveStream = false;
+
+            if (_lastStreamId == -1)
+            {
+                _lastStreamId = lastValidStream;
+            }
+            else
             {
-                if (NetEventSource.IsEnabled) Trace($"{nameof(lastValidStream)}={lastValidStream}, {nameof(abortException)}={lastValidStream}={abortException}");
+                // We have already received GOAWAY before
+                // In this case the smaller valid stream is used
+                _lastStreamId = Math.Min(_lastStreamId, lastValidStream);
+            }
 
-                if (!_disposed)
-                {
-                    shouldInvalidate = true;
-                    _disposed = true;
-                }
+            lock (SyncObject)
+            {
+			    if (NetEventSource.IsEnabled) Trace($"{nameof(lastValidStream)}={lastValidStream}, {nameof(abortException)}={lastValidStream}={abortException}, {nameof(_lastStreamId)}={_lastStreamId}");
```

Comment:
Nit: indentation looks like it's off by a space

Reply:
You have `={lastValidStream}` twice; the second one should be removed.

Reply:
It'd also be good to swap _lastStreamId and abortExceptions, so that the two IDs being output show up next to each other.

Reply:
merge tool :[